agent/consul/leader_connect_ca.go

```diff
@@ -1111,8 +1111,8 @@ func setLeafSigningCert(caRoot *structs.CARoot, pem string) error {
 	return nil
 }
 
-// intermediateCertRenewalWatch periodically attempts to renew the intermediate cert.
-func (c *CAManager) intermediateCertRenewalWatch(ctx context.Context) error {
+// runRenewIntermediate periodically attempts to renew the intermediate cert.
+func (c *CAManager) runRenewIntermediate(ctx context.Context) error {
```

Comment:
This method was not really a "watch". Generally our watch functions block on a channel close, but this is polling at some interval, so I thought a rename would be appropriate.

Reply:
[nit + i'm bad at naming, so maybe disregard]: could we somehow capture the "retry"-ness of the func in its name? Something like `renewIntermediateWithRetries` or `retryRenewIntermediate` or `runRetryRenewIntermediate`?

Reply:
Good idea! The function does both poll and retry if one of the ticks fails. In generally try to call any long running functions `Run` (or `Run<Something>`). 

Maybe `runPeriodicIntermediateCertRotation` ?  If that sounds good I'll probably opt to do the rename in a follow up and rename a couple other things at the same time.